ledger/ledger-api-auth/src/main/scala/com/digitalasset/ledger/api/auth/SupportedJWTPayload.scala

```diff
@@ -0,0 +1,47 @@
+// Copyright (c) 2021 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.daml.ledger.api.auth
+
+import com.daml.ledger.api.auth.AuthServiceJWTCodec.{
+  readPayload,
+  readStandardTokenPayload,
+  writePayload,
+  writeStandardTokenPayload,
+}
+import spray.json.{DefaultJsonProtocol, JsValue, RootJsonFormat}
+
+/** A wrapper class to add support for standard JWT tokens alongside the existing custom tokens in a minimally invasive way.
+  *
+  * The main problem is that the [[AuthServiceJWTPayload]] class is used by other applications
+  * like the JSON-API to parse tokens; and we don't want to meddle with that code as part of the PoC.
+  */
+// TODO (i12049): clarify naming and inline case classes where possible.
+sealed trait SupportedJWTPayload {}
+final case class CustomDamlJWTPayload(payload: AuthServiceJWTPayload) extends SupportedJWTPayload
```

Comment:
Wouldn't `LegacyDamlJWTPayload` fit better here? At least this is my point of view on this..

Reply:
Temporal names like `Legacy` don't work very well for readers who don't have the context. That's why I've chosen the name that this token contains claims that are custom to Daml.

Reply:
This is true too :thinking: OK, with also keeping in mind that the new version uses no custom claims this naming scheme makes sense too. Fine by me then. Only thing keeps bothering me is that both cases are using the same underlying type. Is there really no way to give the `StandardJWTPayload` an own type holding the fields?

Reply:
Yes, I agree. I'll take care of that code smell in #12049. 

I'm intentionally not working on this PR anymore, as I want to keep it quiet for the review by @nmarton-da and get it merged asap, so you guys can start testing on `main`.

Reply:
That makes a lot sense, many thanks for the clarification!